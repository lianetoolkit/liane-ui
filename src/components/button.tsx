import React, { FC, ReactChild } from 'react';
export interface ButtonProps {
  children?: ReactChild;
  type?:
    | 'primary'
    | 'secondary'
    | 'tertiary'
    | 'ghost'
    | 'destructive'
    | 'inverse';
  size?: 'comfortable' | 'compact';
  style?: any;
  className?: any;
  full?: boolean;
  disabled?: boolean;
  onClick?: Function;
}

export const Button: FC<ButtonProps> = ({
  children,
  type,
  size,
  style,
  className,
  full,
  onClick,
  disabled,
}) => {
  const classSizes =
    size === 'compact' ? 'py-1.5 px-3 text-sm' : 'py-3 px-6 text-base';
  const classTypes = {
    primary:
      'bg-purple border border-purple text-white hover:bg-purple-400 active:bg-purple-700',
    secondary:
      'bg-white border border-purple text-purple hover:bg-purpleLight active:bg-purpleLight-75',
    tertiary:
      'bg-white border border-gray text-black hover:bg-gray-50 active:bg-gray-100',
    ghost:
      'bg-transparent  text-purple hover:bg-purpleLight active:bg-purpleLight-75',
    destructive:
      'bg-red border border-red text-white hover:bg-red-400 active:bg-red-700',
    inverse:
      'text-purple bg-white hover:bg-purpleLight active:bg-purpleLight-75',
  };

  return (
    <button
      onClick={(ev) => {
        if (onClick) onClick(ev);
      }}
      disabled={disabled ?? false}
      style={style ?? {}}
      className={`${full ? `w-full inline-block` : ''}	${classSizes} ${
        type && classTypes[type]
      } ${className ?? ''} ${disabled ? 'bg-gray' : ''} rounded font-medium `}
    >
      {children}
    </button>
  );
};
