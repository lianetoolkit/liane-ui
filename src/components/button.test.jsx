import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button } from './button';

describe('Button', () => {
  it('should render correctly', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });
  it('should render button with given props', () => {
    const component = shallow(<Button type={'primary'} size={'compact'} />);
    expect(component).toMatchSnapshot();
  });
  it('should be possible to activate button with Spacebar', () => {
    const component = mount(<Button id="button-test" />);
    component.find('button#button-test').simulate('keydown', { keyCode: 32 });
    expect(component).toMatchSnapshot();
    component.unmount();
  });
  const clickFn = jest.fn();
  it('button click trigger onclick', () => {
    const component = shallow(<Button id="button-test-2" onClick={clickFn} />);
    component.find('button#button-test-2').simulate('click');
    expect(clickFn).toHaveBeenCalled();
  });
  // Check also disabled
});
