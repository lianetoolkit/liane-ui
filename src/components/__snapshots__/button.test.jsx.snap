// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be possible to activate button with Spacebar 1`] = `ReactWrapper {}`;

exports[`Button should render button with given props 1`] = `ShallowWrapper {}`;

exports[`Button should render correctly 1`] = `ShallowWrapper {}`;
